pkg/util/validation/stack_test.go

```diff
@@ -0,0 +1,54 @@
+package validation
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateStackTag(t *testing.T) {
+	t.Run("valid tag", func(t *testing.T) {
+		tags := map[apitype.StackTagName]string{
+			"tag-name": "tag-value",
+		}
+
+		err := ValidateStackTags(tags)
+		assert.NoError(t, err)
+	})
+
+	t.Run("invalid stack tag name", func(t *testing.T) {
+		tags := map[apitype.StackTagName]string{
+			"hello!": "tag-value",
```

Comment:
Might I suggest we add some more examples of invalid strings here?

What about:

- ""
- "something-with-emoji 😄 "
- "something with spaces"
- "something\twith\nescape\nsequences"